src/current/advisories/a99561.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Technical Advisory 99561
+advisory: A-99561
+summary: When adding a new column using the new declarative schema changer, CockroachDB retrieves a unique secondary index mistakenly, assuming it to be a primary index.
+toc: true
+affected_versions: v22.2.x, v23.1.0-v23.1.5
+advisory_date: 2023-07-18
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+CockroachDB automatically creates a primary [index](../v23.1/indexes.html) for each table, which indexes the table's [primary key](../v23.1/primary-key.html), or another unique value when there is no primary key. A secondary index is also created automatically for columns with a [`UNIQUE`](../v23.1/unique.html) constraint. CockroachDB v20.2 and v21.1 (the first two versions to support [`ALTER PRIMARY KEY`](../v21.1/alter-primary-key.html)) contained a bug where the index automatically created in some cases to provide uniqueness for the existing primary key would have its `EncodingType` field marked as `PrimaryIndexEncoding`. This behavior was fixed in v21.2, but left [secondary indexes](../v23.1/schema-design-indexes.html) with primary encoding.
+
+In CockroachDB v22.2.0 and above, the implementation for [`ALTER TABLE`](../v22.2/alter-table.html#add-column) began using the new [declarative schema changer](../v23.1/online-schema-changes.html#declarative-schema-changer) framework. When adding a new column using the new declarative schema changer, a bug caused the incorrect retrieval of the unique secondary index rather than the primary index. As a result, the secondary index is modified to store the new columns rather than the primary index. That is, `scpb.FindPrimaryIndex` expects there to be only one `scpb.FindPrimaryIndex`, but in this case there are two, because the secondary index is incorrectly labeled as a primary one. This causes the new column data to only be present in the secondary index, which could lead to irrecoverable data loss, if the secondary index is deleted by the user. Furthermore, it could lead to inconsistent query results as the primary index does not contain data corresponding to the new column.
```

Comment:
@rimadeodhar Is this sentence necessary? Can we remove?

> That is, `scpb.FindPrimaryIndex` expects there to be only one `scpb.FindPrimaryIndex`, but in this case there are two, because the secondary index is incorrectly labeled as a primary one.

I would say we don't typically refer to lines of CRDB code in docs. It seems like the meaning is there without this. But, let me know if there is a specific reason we should keep this?

Reply:
Yeah, I agree. I think we can remove it.